sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -488,9 +488,24 @@ public void updateIedNameBasedOnLnode(SCL scl) {
                                                                 && Objects.equals(tlNode.getLnInst(), tCompasFlow.getExtReflnInst())
                                                                 && Utils.lnClassEquals(tlNode.getLnClass(), tCompasFlow.getExtReflnClass())
                                                                 && Objects.equals(tlNode.getPrefix(), tCompasFlow.getExtRefprefix()))
+                                                        .filter(tlNode -> {
+                                                            TCompasICDHeader tCompasICDHeader = PrivateUtils.extractCompasPrivate(tlNode, TCompasICDHeader.class)
+                                                                    .orElseThrow(() -> new ScdException(("The substation LNode with following attributes : IedName:%s / LdInst:%s / LnClass:%s / LnInst:%s  " +
+                                                                            "does not contain the needed (COMPAS - ICDHeader) private")
+                                                                            .formatted(tlNode.getIedName(), tlNode.getLdInst(), tlNode.getLnClass().getFirst(), tlNode.getLnInst())));
+                                                            return Objects.equals(tCompasFlow.getFlowSourceIEDType(), tCompasICDHeader.getIEDType())
+                                                                    && Objects.equals(tCompasFlow.getFlowIEDSystemVersioninstance(), tCompasICDHeader.getIEDSystemVersioninstance())
+                                                                    && Objects.equals(tCompasFlow.getFlowSourceIEDredundancy(), tCompasICDHeader.getIEDredundancy());
+                                                        })
                                                         .map(TLNode::getIedName)
                                                         .filter(StringUtils::isNotBlank)
-                                                        .findFirst()
+                                                        .reduce((iedName1, iedName2) -> {
```

Comment:
If you want you can use a more "fonctional" approach :
```java
.reduce(toOneIedName(tCompasFlow, tBay, sclReportItems))
```

```java
/**
     * Add an error in SclReportItem if mutliple IED names are found inside the Bay
     */
    private static BinaryOperator<String> toOneIedName(TCompasFlow tCompasFlow, TBay tBay, List<SclReportItem> sclReportItems) {
        return (iedName1, iedName2) -> {
            sclReportItems.add(SclReportItem.error("", ("Several LNode@IedName ('%s', '%s') are found in the bay '%s' for the following compas-flow attributes : @FlowSourceIEDType '%s' @FlowSourceIEDredundancy '%s' @FlowIEDSystemVersioninstance '%s'").
                    formatted(iedName1, iedName2, tBay.getName(), tCompasFlow.getFlowSourceIEDType(), tCompasFlow.getFlowSourceIEDredundancy(), tCompasFlow.getFlowIEDSystemVersioninstance())));
            return iedName1;
        };
    }
```

Reply:
Yes but the goal isn't to obtain one Ied from the two but only to add an error if there are 2 or more. 
The returned ied isn't important at all because the scd is considered globally false with the error. If you have a better way to make it readable, feel free :)

Reply:
I could rename the function to "checkOnlyOneIed"

Reply:
I did not understand you first response but it does the exact same thing as the code you writ on you own.
you can rename it to "checkOnlyOneIed" if you want but my idea was to read ".reduce(toOneIedName" to make a readable code.

Anyway this was juste a suggestion, you can do what you want

Reply:
I agree that it is more readable with this, I updated to ".reduce(checkOnlyOneIed()"